### ISSUE TITLE
Can I use less GPU for pretraining?

### ISSUE THREAD

**Issue by zhanghm1995** [MCG-NJU/VideoMAE#20]
Hi, thank you for sharing this amazing work.

I know you have used 64 GPUs spreading in 8 nodes for pretraining. However, I don't have so many GPU resources, I wonder whether I could pretrain this model in 8 GPUs or less.

Whether there are large margin performance decrease?

**Comment by JiayuZou2020** [comment]
If you want to use 6 GPUs for pretraining or fine-tuning, you can run the following code.
`python -m torch.distributed.launch --nproc_per_node=6 --master_port=12320 run_mae_pretraining.py --data_path ${DATA_PATH}  --mask_type tube --mask_ratio 0.9 --model pretrain_videomae_base_patch16_224 --decoder_depth 4  --batch_size 32 --num_frames 16 --sampling_rate 4 --opt adamw --opt_betas 0.9 0.95 --warmup_epochs 40  --save_ckpt_freq 20 --epochs 801 --log_dir ${OUTPUT_DIR} --output_dir ${OUTPUT_DIR}`

**Comment by jianyulei1** [comment]
Hi, if I want to set batch size to 8, does LR need to be changed?

**Comment by mjlbach** [comment]
I've had issues training with less GPUs (6 v100s), training on SSV2 we get about 40% higher loss than with the pretrained model (from the log files) at 800 epochs. The LR is scaled by num_training_steps per epoch which depends on the number of tasks.

Does anyone have any suggestions?

**Comment by Zi-hao-Wei** [comment]
In my case, I added an accumlate iter arguments during training, which can scale up the effective batch size. The accum_iter's implementation can be found in the repo of MAE.

**Comment by mjlbach** [comment]
@Zi-hao-Wei Would you mind sharing your modifications/code?

**Comment by Zi-hao-Wei** [comment]
You can find it here https://github.com/Zi-hao-Wei/VideoMAE_acc_iter; Maybe I can add a pull/request to the original repo?

**Comment by mjlbach** [comment]
@Zi-hao-Wei Thanks! I think this would be a good addition to the repo (not maintainer ofc), however, have you managed to reproduce the quality of the reconstructions? I'm going to try running on 8 V100's this week and see what it looks like on SSV2

**Comment by mjlbach** [comment]
@Zi-hao-Wei Were you able to get good results with this? When I tried setting accum_iter to 8 on 8 A40s the loss seemed to not improve (green: accum_iter = 8, vs. grey: accum_iter=1)

<img width="380" alt="image" src="https://user-images.githubusercontent.com/13316262/191132629-3130552d-e310-4e65-a1dc-fa923878aa0c.png">

**Comment by Zi-hao-Wei** [comment]
Thank you for sharing. I will try to follow your implementation.

**Comment by congee524** [comment]
too small batch size may degrade the model performance, maybe you can use checkpoint.checkpoint to save memory.
